packages/Caching/ValueObject/Storage/FileCacheStorage.php

```diff
@@ -84,10 +84,11 @@ public function clean(string $cacheKey): void
         $cacheFilePaths = $this->getCacheFilePaths($cacheKey);
 
         $this->smartFileSystem->remove([
-            $cacheFilePaths->getFirstDirectory(),
-            $cacheFilePaths->getSecondDirectory(),
             $cacheFilePaths->getFilePath(),
         ]);
+        // remove dirs only if empty to prevent over aggressive delete
+        @rmdir($cacheFilePaths->getSecondDirectory());
+        @rmdir($cacheFilePaths->getFirstDirectory());
```

Comment:
Better avoid `@` magic, that will bite us sooner or later

Reply:
@staabm In that case we cannot use rmdir and glob or scandir must be used.

Reply:
I have checked phpstan caching and they don't remove that at all, they just returns invalid cache.

Reply:
I think it does not make sense to remove that folders, we got that file, which exist in code base and it is invalid and we are sure that cache file will be created again with need checksum, on at same place, there is totally no need for removing that folders.

Reply:
Ok fine, kill it :-)

Reply:
Ok, I have removed single cache clean completely, since it is not needed, invalid cache files are overridden later in code.